makePoplogCommander.sh

```diff
@@ -25,6 +27,92 @@ cat << \****
 #define PREFER_FLEXIBILITY  (RUN_INIT_P|INHERIT_ENV)
 
 
+///////////////////////////////////////////////////////////////////////////////
+//  Library routines: 
+//      Mishap, an error reporting function
+//      Chain, a 1D vector
+///////////////////////////////////////////////////////////////////////////////
+
+
+//  Mishap - Error reporting using printf-like functionality.
+
+void mishap( const char *msg, ... ) {
+    va_list args;
+    va_start( args, msg );
+    fprintf( stderr, "Mishap: " );
+    vfprintf( stderr, msg, args );
+    fprintf( stderr, "\n" );
+    va_end( args );
+    exit( EXIT_FAILURE );
+}
+
+//  Chains - 1D vectors.
+
+typedef struct Chain *Chain;
+typedef char * Ref;
+
+enum {
+    BUMP = 16
+};
+
+struct Chain {
+    int     size;
+    int     used;
+    Ref     *data;
+};
+
+// Ensure there is room for at least n more bytes
+// in the chain's buffer.
+//
+static Chain bump( Chain r, int n ) {
+    int size = r->size;
+    int used = r->used;
+    int newused = used + n;
+    if ( newused > size ) {
+        int newsize = newused + BUMP;
+        r->data = (Ref *)realloc( r->data, newsize * sizeof( Ref ) );
+        r->size = newsize;
+    }
+    return r;
+}
+
+Chain chain_push( Chain r, Ref ch ) {
+    bump( r, 1 );
+    r->data[ r->used ] = ch;
+    r->used += 1;
+    return r;
+}
+
+Chain chain_new() {
+    Chain r;
+    r = (Chain)malloc( sizeof( struct Chain ) );
+    r->size = 0;
+    r->used = 0;
+    r->data = NULL;
```

Comment:
Can this not be simplified by use of `calloc` instead? Would the initialisation logic not go away?

Reply:
That's a sensible idea. I will amend it.

Reply:
Fixed.